app/form_objects/support/new_user_school_form.rb

```diff
@@ -9,8 +11,10 @@ def initialize(user:, name_or_urn: nil)
   end
 
   def matching_schools
-    School.includes(:responsible_body)
-          .where('urn = ? OR name ILIKE(?)', @name_or_urn.to_i, "%#{@name_or_urn}%")
-          .order(:name)
+    School
+      .matching_name_or_urn(@name_or_urn)
+      .includes(:responsible_body)
+      .order(:name)
+      .limit(MAX_NUMBER_OF_SUGGESTED_SCHOOLS)
```

Comment:
I noticed that we dont have any unit tests for this form_object.r5

Reply:
I've added coverage to f4c220a and force-pushed.